encfs/main.cpp

```diff
@@ -768,9 +772,18 @@ static bool unmountFS(EncFS_Context *ctx) {
     return false;
   } else {
     // Time to unmount!
-    RLOG(WARNING) << "Unmounting filesystem due to inactivity: "
-                  << arg->opts->mountPoint;
-    fuse_unmount(arg->opts->mountPoint.c_str());
-    return true;
+    int rc=1;
+    if (access("/bin/umount", F_OK) != -1)
+      rc = system(("/bin/umount "+std::string(arg->opts->mountPoint)).c_str());
+    else
+      rc = system(("/sbin/umount "+std::string(arg->opts->mountPoint)).c_str());
```

Comment:
Thank you for your review Sam !

The risk of searching through `$PATH` is to run a malicious `umount`, this is why I used only hard-coded paths.

FreeBSD :
```
# which umount
/sbin/umount
```

Mac OS :
```
# which umount
/sbin/umount
```

Do you have some systems in mind with `umount` elsewhere ?

Regarding Windows, good question.
Seems that encfs4win [calls](https://github.com/jetwhiz/encfs4win/blob/v1.10.1-RC11/encfs/main.cpp#L957) `fuse_unmount` but [disables](https://github.com/jetwhiz/encfs4win/blob/v1.10.1-RC11/encfs/main.cpp#L51) `fuse_unmount_compat22`, as this `fuse_unmount` call goes to [Dokany](https://github.com/dokan-dev/dokany/tree/v1.0.3) library.